src/platform/DeviceControlServer.cpp

```diff
@@ -54,12 +57,16 @@ void DeviceControlServer::CommissioningFailedTimerComplete()
 
 CHIP_ERROR DeviceControlServer::ArmFailSafe(System::Clock::Timeout expiryLength)
 {
+    mFailSafeArmed = true;
```

Comment:
1. This could use some documentation in the header about how the caller must ensure that it's safe to arm the failsafe (i.e. that no conflicting failsafe is armed).
2. This could use some documentation in the impl about how if we're already armed we just reset the timer, which is what StartTimer ends up doing if the timer is already started.
3. If we are called with `mFailSafeArmed == false`, we should store the accessing fabric index in the failsafe context and set its "have seen NOC command" boolean to false.

Reply:
mFailSafeArmed == false is called either when DisarmFailSafe or CommissioningComplete are called, we should reset mFabricIndex to kUndefinedFabricIndex, right? why we need to still track the accessing fabric index